scrapy/loader/__init__.py

```diff
@@ -113,7 +113,7 @@ def load_item(self):
         item = self.item
         for field_name in tuple(self._values):
             value = self.get_output_value(field_name)
-            if value is not None:
+            if value is not None and value != []:
```

Comment:
I can imagine cases where someone _expects_ a failed load to still populate an empty list, and this change might break things for them. Perhaps instead the code should check whether adding an empty list would clobber an existing field? Because, per your issue in #3046 I think that's the more bug-like behaviour?